### PR TITLE
Fix SqlDataReader.IsDBNull() for json

### PR DESCRIPTION
**Pull request by deepaksa1**
This PR aims to fix `SqlDataReader.IsDBNull()` API for json data type.
Currently, it returns false for a null json column. Adding the case for json type returns expected output.

### CODE REVIEW

**Comment by ErikEJ** [comment]
Possible to add a test?

**Review by apoorvdeshmukh — COMMENTED** [review]
Looks good. Please add a testcase that fails without your fix.

**Review by saurabh500 — APPROVED** [review]

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonTest.cs [hunk 1]
```diff
@@ -286,5 +296,40 @@ public async Task TestJsonReadAsync()
...
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.IsJsonSupported))]
+        public void TestNullJson()
+        {
+            string tableName = DataTestUtility.GetUniqueNameForSqlServer("Json_Test");
+
+            string tableCreate = "CREATE TABLE " + tableName + " (Data json)";
+            string tableInsert = "INSERT INTO " + tableName + " VALUES (@jsonData)";
+            string tableRead = "SELECT * FROM " + tableName;
+
+            using (SqlConnection connection = new SqlConnection(DataTestUtility.TCPConnectionString))
```
Comment (roji) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonTest.cs:309]: Consider using modern C# constructs:

```c#
using SqlConnection connection = new SqlConnection(DataTestUtility.TCPConnectionString);
```

This reduces the unneeded nesting etc.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonTest.cs [hunk 2]
```diff
@@ -286,5 +296,40 @@ public async Task TestJsonReadAsync()
...
+            {
+                connection.Open();
+                using (SqlCommand command = connection.CreateCommand())
+                {
+                    //Create Table
+                    command.CommandText = tableCreate;
```
Comment (apoorvdeshmukh) [inline src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonTest.cs:315]: Consider using DropTable() or CreateTable() function from DataTestUtility to ensure tables created by test are dropped properly.

**Review by apoorvdeshmukh — APPROVED** [review]
Suggested few minor changes.

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonTest.cs (hunk 1 above)
Comment (deepaksa1) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonTest.cs]: modified

File: src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonTest.cs (hunk 2 above)
Comment (deepaksa1) [reply src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/JsonTest/JsonTest.cs]: added the code for drop table

**Pull request merged**